src/Admin/SiteHealth.php

```diff
@@ -385,35 +387,115 @@ public function page_cache() {
 	 *
 	 * @param bool $use_previous_result Whether to use previous result or not.
 	 *
-	 * @return bool|WP_Error Boolean if the site has page caching or not, or else a WP_Error if unable to determine.
+	 * @return string|WP_Error Boolean if the site has page caching or not, or else a WP_Error if unable to determine.
 	 */
-	public function has_page_caching( $use_previous_result = false ) {
+	public function get_page_cache_status( $use_previous_result = false ) {
 
 		if ( $use_previous_result ) {
-			$has_page_caching = get_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY );
-			if ( is_wp_error( $has_page_caching ) ) {
-				return $has_page_caching;
-			} elseif ( $has_page_caching ) {
-				return ( 'yes' === $has_page_caching );
+			$page_cache_detail = get_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY );
+		}
+
+		if ( ! $use_previous_result || empty( $page_cache_detail ) ) {
+			$page_cache_detail = $this->check_for_page_caching();
+			if ( is_wp_error( $page_cache_detail ) ) {
+				set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $page_cache_detail, DAY_IN_SECONDS );
+			} else {
+				set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $page_cache_detail, MONTH_IN_SECONDS );
 			}
 		}
 
-		$has_page_caching = $this->check_for_page_caching();
-		if ( is_wp_error( $has_page_caching ) ) {
-			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching, DAY_IN_SECONDS );
-		} else {
-			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching ? 'yes' : 'no', MONTH_IN_SECONDS );
+		if ( is_wp_error( $page_cache_detail ) ) {
+			return $page_cache_detail;
+		}
+
+		if ( is_string( $page_cache_detail ) ) {
+			return 'yes' === $page_cache_detail ? 'good' : 'critical';
+		}
+
+		$page_speed            = array_sum( $page_cache_detail['response_timing'] ) / count( $page_cache_detail['response_timing'] );
```

Comment:
This is getting the average response time. That's one possibility. Two other possibilities are to get the _minimum_ response time and the other is to get the _median_ response time. It could be that the first request could be uncached, so it could have a longer response time. If so, then averaging the responses would mean that first response would negatively impact the overall page speed we obtain. However, if we get the median then the first long request would be essentially thrown away and we'd only consider the middle response of the 3. That may be more insigtful.

```suggestion
		$response_timings = $page_cache_detail['response_timing'];
		rsort( $response_timings );
		$page_speed            = $response_timings[ floor( count( $response_timings ) / 2 ) ];
```